### DISCUSSION TITLE
T-Pot 24.04.0 released 🎉

### DISCUSSION THREAD

**Discussion by t3chn0m4g3** [telekom-security/tpotce#1519] [Announcements]
# Release Notes / Changelog
T-Pot 24.04.0 marks probably the largest change in the history of the project. While most of the changes have been made to the underlying platform some changes will be standing out in particular - a T-Pot ISO image will no longer be provided with the benefit that T-Pot will now run on multiple Linux distributions (Alma Linux, Debian, Fedora, OpenSuse, Raspbian, Rocky Linux, Ubuntu), Raspberry Pi (optimized) and macOS / Windows (limited).

## New Features
* **Distributed** Installation is now using NGINX reverse proxy instead of SSH to transmit **HIVE_SENSOR** logs to **HIVE**
* **`deploy.sh`**, will make the deployment of sensor much easier and will automatically take care of the configuration. You only have to install the T-Pot sensor.  
* **T-Pot Init** is the foundation for running T-Pot on multiple Linux distributions and will also ensure to restart containers with failed healthchecks using **autoheal**
* **T-Pot Installer** is now mostly Ansible based providing a universal playbook for the most common Linux distributions 
* **T-Pot Uninstaller** allows to uninstall T-Pot, while not recommended for general usage, this comes in handy for testing purposes
* **T-Pot Customizer (`compose/customizer.py`)** is here to assist you in the creation of a customized `docker-compose.yml`
* **T-Pot Landing Page** has been redesigned and simplified
* **Kibana Dashboards, Objects** fully refreshed in favor of Lens based objects
* **Wordpot** is added as new addition to the available honeypots within T-Pot and will run on `tcp/8080` by default.
* **Raspberry Pi** is now supported using a dedicated `mobile.yml` (why this is called mobile will be revealed soon!)
* **GeoIP Attack Map** is now aware of connects / disconnects and thus eliminating required reloads
* **Docker**, where possible, will now be installed directly from the Docker repositories to avoid any incompatibilities
* **`.env`** now provides a single configuration file for the T-Pot related settings
* **`genuser.sh`** can now be used to add new users to the T-Pot Landing Page as part of the T-Pot configuration file (`.env`)

## Updates
* **Honeypots** and **tools** were updated to their latest pushed code and / or releases
* Where possible Docker Images will now use Alpine 3.19
* Updates will be provided continuously through Docker Images updates 

## Breaking Changes
* There is no option to migrate a previous installation to T-Pot 24.04.0, you can try to transfer the old `data` folder to the new T-Pot installation, but a working environment depends on too many other factors outside of our control and a new installation is simply faster.
* Most of the support scripts were moved into the **T-Pot Init** image and are no longer available directly on the host.
* Cockpit is no longer available as part of T-Pot itself. However, where supported, you can simply install the `cockpit` package.

# Thanks & Credits
* @sp3t3rs, @trixam, for their backend and ews support!
* @cha147 made their first contribution in https://github.com/telekom-security/tpotce/pull/1135
* @ctulio made their first contribution in https://github.com/telekom-security/tpotce/pull/1187
* @zambroid made their first contribution in https://github.com/telekom-security/tpotce/pull/1173
* @kawaiipantsu made their first contribution in https://github.com/telekom-security/tpotce/pull/1259
* @tadashi-oya made their first contribution in https://github.com/telekom-security/tpotce/pull/1283
* @kauedg made their first contribution in https://github.com/telekom-security/tpotce/pull/1338
* @swiftsolves-msft made their first contribution in https://github.com/telekom-security/tpotce/pull/1369
* @shark4ce for taking the time to test, debug and offer a solution #1472

## History of merged changes
* Fix typos in readme by @cha147 in https://github.com/telekom-security/tpotce/pull/1135
* Update some url repos by @ctulio in https://github.com/telekom-security/tpotce/pull/1187
* Corrected small typos by @zambroid in https://github.com/telekom-security/tpotce/pull/1173
* Update updateip.sh by @kawaiipantsu in https://github.com/telekom-security/tpotce/pull/1259
* Fixing uri max size by @kawaiipantsu in https://github.com/telekom-security/tpotce/pull/1266
* fix empty myINSTALLPACKAGES by @tadashi-oya in https://github.com/telekom-security/tpotce/pull/1283
* call $0 instead of hardcoded script name by @kauedg in https://github.com/telekom-security/tpotce/pull/1338
* fixes #1346 by @t3chn0m4g3 in https://github.com/telekom-security/tpotce/pull/1351
* Azure Deployment via ARM template by @swiftsolves-msft in https://github.com/telekom-security/tpotce/pull/1369
* 24.04 by @t3chn0m4g3 in https://github.com/telekom-security/tpotce/pull/1513

**Full Changelog**: https://github.com/telekom-security/tpotce/compare/22.04.0...24.04.0

<hr /><em>This discussion was created from the release <a href='https://github.com/telekom-security/tpotce/releases/tag/24.04.0'>24.04.0</a>.</em>

**Comment by aishla** [comment]
Hi, I'm having a problem installing T-Pot and I haven't been able to fix it even though I've tried installing it several times. I always get the same error. How can I fix it?
![unnamed](https://github.com/telekom-security/tpotce/assets/100279871/75f32155-f8e2-4e36-8f27-5473818aa4f4)

**Reply by t3chn0m4g3** [reply]
You are trying to install a previous version of T-Pot. Please follow the instructions of the Readme for T-Pot 24.04.

**Comment by msaleh00** [comment]
how can i get .iso file ?

**Reply by aishla** [reply]
there is a new version on t-pot, the instalation of this version doesn't work

**Comment by kanowah** [comment]
How to prevent the server from restarting everyday? Can't find it in the crontab anymore!

**Reply by t3chn0m4g3** [reply]
Use `sudo crontab -e` for editing the crontab and `sudo crontab -l` for viewing it.

**Reply by kanowah** [reply]
I know these steps, but I can't find the line that need to be removed to prevent daily reboot in the newest version.

**Reply by t3chn0m4g3** [reply]
Here is an [example](https://github.com/telekom-security/tpotce?tab=readme-ov-file#daily-reboot).

**Reply by kanowah** [reply]
Thank you

**Comment by ShoaibTalib** [comment]
I have been trying up for so long to install tpot 24.04.0 (latest) on Debian 12 . i am having some errors , i tried ubuntu server also .. attached screen shot is for reference 
![image (1)](https://github.com/telekom-security/tpotce/assets/71877489/1bfc5664-37b4-4163-81b2-00c10bc61f1f)
![image](https://github.com/telekom-security/tpotce/assets/71877489/2089c4c7-399f-49d9-89aa-4006835ecd58)
 
Kindly look into this grievance.  Secondly it would be better to make iso for this version , as like in 22.04 verrsion

**Reply by t3chn0m4g3** [reply]
You are trying to install as root and you are not in the $HOME folder.
Follow the ReadMe for all the necessary instructions for a successful installation.

**Comment by ghenry** [comment]
You need to install ansible-playbook like the error indicates 😎

**Reply by ShoaibTalib** [reply]
i tried this also , but after running there are other error that i am getting

**Reply by ShoaibTalib** [reply]
You can see the errors in second screen shot of ubuntu server !!!

**Reply by t3chn0m4g3** [reply]
From the looks of it you are trying to install on top of the Desktop edition of either Ubuntu or Debian. Please follow the ReadMe and use a supported distribution ISO and install no extra packages, except for SSH.

**Reply by ShoaibTalib** [reply]
![image (2)](https://github.com/telekom-security/tpotce/assets/71877489/913e1b67-ee34-4905-85f1-6693910e48e3)
 i deployed the tpot successfully on ubuntu server , and i checked either tpot is running or not using sudo systemctl status tpot, now my question is how can i run elastic search kibana dashboard, like previously in tpot 22.02 , it was easy just to use port no with external ip and we were good to go , but how can i run on this  , secondly how can i perform attacks on cowrie using kali linux, i have deployed both vms in GNS3 software and connected via router , my research is basically on honeypots , kindly guide me ..

**Comment by kanowah** [comment]
I have an issue with this new version, after 24 hours after the installation, I'm getting this message and the attack map stop working, how to fix this issue?
![Screenshot 2024-05-27 085217](https://github.com/telekom-security/tpotce/assets/72765704/d9320a2b-243e-45a2-8ee3-bef9f527c8fa)

**Reply by ShoaibTalib** [reply]
how did you access this dashboard, which ip ?? kindly give me some details

**Reply by kanowah** [reply]
Accessing from Kibana > T-POT Dashboard, after 24 hours after the installation I'm getting these error messages. I've done some research trying to solve this issue but I'm still stuck, the only solution that is working for me is to reinstall everything again, but same error appear after 24 hours.

**Reply by ShoaibTalib** [reply]
i mean to say on which port kibana is running , how can i access my kibana dashboard, what ip would i be using ? Local ip address starting off with 192.168 .. , i have deployed this on ubuntu server running on vmware, which is put in bridge connection  kindly respond

**Comment by pwner-jw** [comment]
Hey, i am having an issue with the Tpot dashboard, i have tried installing Tpot on Ubuntu 23.10 and RockyLinux9 and in both of my installation, i could not access the web interface, or even if i could, not ever part of it was working

Also i am new to TPot, is there any guide about all the available, configs and other stuff that we can refer too
Thankyou

**Comment by ARGUS269** [comment]
Hello, can you give me all the information about the system you used to download t-pot

**Reply by atbohmer** [reply]
Works as a charm on a Debian GNU/Linux 12 (bookworm) VM, thanks for the great work!

**Reply by ARGUS269** [reply]
Thank you, I am having difficulty with tpot, did you use commands like "docker build" and "docker-compose up" to configure the honeypot?I have a report dated 6/26
